plugins/hls-call-hierarchy-plugin/src/Ide/Plugin/CallHierarchy/Query.hs

```diff
@@ -19,55 +20,75 @@ import           Name
-            \AND \
-            \(defs.occ != ? OR rm.mod != ? OR rm.unit != ?) \
-            \AND \
-            \((refs.sl = decls.sl AND refs.sc >  decls.sc) OR (refs.sl > decls.sl)) \
-            \AND \
-            \((refs.el = decls.el AND refs.ec <= decls.ec) OR (refs.el < decls.el))" (o, m, u, o, m, u)
+    let (o, m, u) = parseSymbol symbol
+    query conn
+        (Query $ T.pack $ concat
+            [ "SELECT rm.mod, defs.occ, rm.hs_src, defs.sl, defs.sc, defs.el, defs.ec, "
+            , "refs.sl, refs.sc, refs.el, refs.ec "
+            , "from refs "
+            , "JOIN defs ON defs.occ = refs.occ "
+            , "JOIN decls rd ON rd.hieFile = defs.hieFile AND rd.occ = defs.occ "
+            , "JOIN mods rm ON rm.mod = refs.mod AND rm.unit = refs.unit AND rm.hieFile = defs.hieFile "
+            , "JOIN decls ON decls.hieFile = refs.hieFile "
+            , "JOIN mods ON mods.hieFile = decls.hieFile "
+            , "where "
+            , "(decls.occ = ? AND mods.mod = ? AND mods.unit = ?) "
+            , "AND "
+            , "(defs.occ != ? OR rm.mod != ? OR rm.unit != ?) "
+            , "AND "
+            , "((refs.sl = decls.sl AND refs.sc >  decls.sc) OR (refs.sl > decls.sl)) "
+            , "AND "
+            , "((refs.el = decls.el AND refs.ec <= decls.ec) OR (refs.el < decls.el))"
+            ]
+        ) (o, m, u, o, m, u)
 
 getSymbolPosition :: HieDb -> Vertex -> IO [SymbolPosition]
 getSymbolPosition (getConn -> conn) Vertex{..} = do
-  query conn "SELECT refs.sl, refs.sc from refs where \
-             \(occ = ?) \
-             \AND \
-             \((refs.sl = ? AND refs.sc > ?) OR (refs.sl > ?)) \
-             \AND \
-             \((refs.el = ? AND refs.ec <= ?) OR (refs.el < ?))"
-             (occ, sl, sc, sl, el, ec, el)
+    query conn
+        (Query $ T.pack $ concat
+            [ "SELECT refs.sl, refs.sc from refs where "
+            , "(occ = ?) "
+            , "AND "
+            , "((refs.sl = ? AND refs.sc > ?) OR (refs.sl > ?)) "
+            , "AND "
+            , "((refs.el = ? AND refs.ec <= ?) OR (refs.el < ?))"
+            ]
+        ) (occ, sl, sc, sl, el, ec, el)
 
+#if MIN_VERSION_ghc(9,0,0)
 parseSymbol :: Symbol -> (String, String, String)
 parseSymbol Symbol{..} =
-  let o = toNsChar (occNameSpace symName) : occNameString symName
-      m = moduleNameString $ moduleName symModule
-      u = unitIdString $ moduleUnitId symModule
-  in  (o, m, u)
+    let o = toNsChar (occNameSpace symName) : occNameString symName
+        m = moduleNameString $ moduleName symModule
+        u = unitString $ moduleUnit symModule
+    in  (o, m, u)
+#else
+parseSymbol :: Symbol -> (String, String, String)
+parseSymbol Symbol{..} =
+    let o = toNsChar (occNameSpace symName) : occNameString symName
+        m = moduleNameString $ moduleName symModule
+        u = unitIdString $ moduleUnitId symModule
+    in  (o, m, u)
+#endif
```

Comment:
Avoid using CPP here, instead look to reuse one of the helpers defined in the module `Development.IDE.GHC.Compat` specifically for this purpose.

Reply:
The helper in this case is called simply `moduleUnitId` and it does the right thing for all the supported GHC versions